### ISSUE TITLE
Getting this even though project has access to that model

### ISSUE THREAD

**Issue by surapuramakhil** [feder-cr/linkedIn_auto_jobs_applier_with_AI#289]
Traceback (most recent call last):
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/src/linkedIn_easy_applier.py", line 60, in job_apply
    self.gpt_answerer.set_job(job)
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/src/gpt.py", line 147, in set_job
    self.job.set_summarize_job_description(self.summarize_job_description(self.job.description))
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/src/gpt.py", line 158, in summarize_job_description
    output = chain.invoke({"text": text})
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 2875, in invoke
    input = step.invoke(input, config)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 4441, in invoke
    return self._call_with_config(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 1784, in _call_with_config
    context.run(
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/config.py", line 404, in call_func_with_variable_args
    return func(input, **kwargs)  # type: ignore[call-arg]
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 4297, in _invoke
    output = call_func_with_variable_args(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/config.py", line 404, in call_func_with_variable_args
    return func(input, **kwargs)  # type: ignore[call-arg]
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/src/gpt.py", line 86, in __call__
    reply = self.llm(messages)
            ^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/_api/deprecation.py", line 168, in warning_emitting_wrapper
    return wrapped(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 976, in __call__
    generation = self.generate(
                 ^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 571, in generate
    raise e
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 561, in generate
    self._generate_with_cache(
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 793, in _generate_with_cache
    result = self._generate(
             ^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_openai/chat_models/base.py", line 589, in _generate
    response = self.client.create(**payload)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/openai/_utils/_utils.py", line 277, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/openai/resources/chat/completions.py", line 646, in create
    return self._post(
           ^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/openai/_base_client.py", line 1266, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/openai/_base_client.py", line 942, in request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/openai/_base_client.py", line 1046, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.PermissionDeniedError: Error code: 403 - {'error': {'message': 'Project `proj_nX3lgCQHQmUiWJG6YzQNFcPz` does not have access to model `gpt-4o-mini`', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/src/linkedIn_job_manager.py", line 126, in apply_jobs
    self.easy_applier_component.job_apply(job)
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/src/linkedIn_easy_applier.py", line 65, in job_apply
    raise Exception(f"Failed to apply to job! Original exception: \nTraceback:\n{tb_str}")
Exception: Failed to apply to job! Original exception: 
Traceback:
Traceback (most recent call last):
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/src/linkedIn_easy_applier.py", line 60, in job_apply
    self.gpt_answerer.set_job(job)
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/src/gpt.py", line 147, in set_job
    self.job.set_summarize_job_description(self.summarize_job_description(self.job.description))
                                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/src/gpt.py", line 158, in summarize_job_description
    output = chain.invoke({"text": text})
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 2875, in invoke
    input = step.invoke(input, config)
            ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 4441, in invoke
    return self._call_with_config(
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 1784, in _call_with_config
    context.run(
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/config.py", line 404, in call_func_with_variable_args
    return func(input, **kwargs)  # type: ignore[call-arg]
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/base.py", line 4297, in _invoke
    output = call_func_with_variable_args(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/runnables/config.py", line 404, in call_func_with_variable_args
    return func(input, **kwargs)  # type: ignore[call-arg]
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/src/gpt.py", line 86, in __call__
    reply = self.llm(messages)
            ^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/_api/deprecation.py", line 168, in warning_emitting_wrapper
    return wrapped(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 976, in __call__
    generation = self.generate(
                 ^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 571, in generate
    raise e
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 561, in generate
    self._generate_with_cache(
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_core/language_models/chat_models.py", line 793, in _generate_with_cache
    result = self._generate(
             ^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/langchain_openai/chat_models/base.py", line 589, in _generate
    response = self.client.create(**payload)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/openai/_utils/_utils.py", line 277, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/openai/resources/chat/completions.py", line 646, in create
    return self._post(
           ^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/openai/_base_client.py", line 1266, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/openai/_base_client.py", line 942, in request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/Users/akhilsurapuram/openSource/linkedIn_auto_jobs_applier_with_AI/virtual/lib/python3.11/site-packages/openai/_base_client.py", line 1046, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.PermissionDeniedError: Error code: 403 - {'error': {'message': 'Project `proj_nX3lgCQHQmUiWJG6YzQNFcPz` does not have access to model `gpt-4o-mini`', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}

**Comment by 50sotero** [comment]
try using a new api key

**Comment by Carl0-C0des** [comment]
I had this issue occur - if you just created your Open API account and are on the free tier, you do not get access to GPT-4o-mini as stated in the last line of the error: "openai.PermissionDeniedError: Error code: 403 - {'error': {'message': 'Project proj_nX3lgCQHQmUiWJG6YzQNFcPz does not have access to model gpt-4o-mini', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}" 

go to open api account make sure that you are in the proper organization and project - then go to Dashboard --> API Keys --> Click "View User API Keys" on the left-hand panel will be "Organization" select "Limits" underneath org. Scroll all the way down and it should indicate which tier your account is considered. Even if you loaded credits $$ over the $5 threshold Open AI has a bug on their side that is failing to automatically upgrade new accounts to Tier 1 as of recently. 

I had to Click "help" then open a new message with their internal support team providing screenshots of my account tier being incorrect. They fixed my account after about a day or so and then my scripts ran fine.

**Comment by Carl0-C0des** [comment]
> I had this issue occur - if you just created your Open API account and are on the free tier, you do not get access to GPT-4o-mini as stated in the last line of the error: "openai.PermissionDeniedError: Error code: 403 - {'error': {'message': 'Project proj_nX3lgCQHQmUiWJG6YzQNFcPz does not have access to model gpt-4o-mini', 'type': 'invalid_request_error', 'param': None, 'code': 'model_not_found'}}"
> 
> go to open api account make sure that you are in the proper organization and project - then go to Dashboard --> API Keys --> Click "View User API Keys" on the left-hand panel will be "Organization" select "Limits" underneath org. Scroll all the way down and it should indicate which tier your account is considered. Even if you loaded credits $$ over the $5 threshold Open AI has a bug on their side that is failing to automatically upgrade new accounts to Tier 1 as of recently.
> 
> I had to Click "help" then open a new message with their internal support team providing screenshots of my account tier being incorrect. They fixed my account after about a day or so and then my scripts ran fine.

TLDR;, open Api considers new accounts "Free Tier" which only gives you access to GPT-3 etc unless you load $5 into your new account credit balance. Even after doing so, there is a current bug that doesn't upgrade your new account from 'free' to "Tier 1" which will give you access to the proper GPT-4 and 4o mini models that this repo requires to execute

**Comment by yulong-lan** [comment]
@Carl0-C0des I understand this error occurs when the user does not have access to gpt-4.0-mini. But where do we change the python code to switch to a different gpt model, such as a free gpt-3.5 model? Must we use a payable gpt model?

**Comment by yulong-lan** [comment]
Ok I tried and it didn't work:

```
    raise self._make_status_error_from_response(err.response) from None
openai.RateLimitError: Error code: 429 - {'error': {'message': 'You exceeded your current quota, please check your plan and billing details. For more information on this error, read the docs: https://platform.openai.com/docs/guides/error-codes/api-errors.', 'type': 'insufficient_quota', 'param': None, 'code': 'insufficient_quota'}}
```
And this [429 error](https://platform.openai.com/docs/guides/error-codes/api-errors) is about: 

```
429 - Rate limit reached for requests
This error message indicates that you have hit your assigned rate limit for the API. This means that you have submitted too many tokens or requests in a short period of time and have exceeded the number of requests allowed. This could happen for several reasons, such as:

You are using a loop or a script that makes frequent or concurrent requests.
You are sharing your API key with other users or applications.
You are using a free plan that has a low rate limit.
You have reached the defined limit on your project
```

I was naive...

**Comment by feder-cr** [comment]
@surapuramakhil @50sotero @yulong-lan @Carl0-C0des join the telegram group, so we can discuss it more easily - > https://t.me/AIhawkCommunity

**Comment by surapuramakhil** [comment]
Getting a new API token has solved this issue.

**Comment by barefoot-dave** [comment]
I am tier 1 and have funds loaded. I created a new key as well. Still getting same error.
